src/scripts/surf.ts

```diff
@@ -0,0 +1,74 @@
+// Description:
+//  Surf server stat comparison
+//
+// Commands:
+//  hubot surf me - compare surfers
+//
+// Author:
+//  Steve Shipsey
+
+import Axios, { AxiosResponse } from 'axios';
+import * as Cheerio from 'cheerio';
+import { Response, Robot } from 'hubot';
+import { table, TableUserConfig } from 'table';
+
+const baseUrl = 'https://snksrv.com/surfstats';
+const surfers = [
+  'STEAM_1:0:6030042',
+  'STEAM_1:0:1166568',
+];
+const headers = ['Surfer', 'Points', 'Map Completions'];
+
+const fetchSurfer = async (steamId: string) => await Axios.get<any>(`${baseUrl}`, {
+  params: {
+    id: steamId,
+    view: 'profile',
+  },
+});
+
+const parse = (cheerio: any): string[] => {
+  const $ = Cheerio.load(cheerio.data);
+  return [
+    $('h2').first().text(),
+    getTextWithoutChildren($('td').first()).slice(2),
+    getTextWithoutChildren($('td').eq(2)).slice(2),
+  ];
+};
+
+const sortByPoints = (surfDataColumns: string[][]) =>
+  surfDataColumns.sort((a, b) => parseInt(b[1], 10) - parseInt(a[1], 10));
+
+const getTextWithoutChildren = (cheerio: Cheerio) => cheerio.clone()
+  .children()
+  .remove()
+  .end()
+  .text();
+
+const zip = <T>(...arrays: T[][]) => {
+  const length = Math.min(...arrays.map(arr => arr.length));
+  return Array.from({ length }, (value, index) => arrays.map((array => array[index])));
+};
+
+const calcTableFormat = (columns: string[][]): TableUserConfig => ({
+  columns: columns.map((column, i) => ({
+    [i]: { width: Math.max(...column.map(c => c.length)) },
+  })),
+});
+
+const parseAndSortResponse = (res: AxiosResponse<any>[]) => sortByPoints(res.map(c => parse(c)));
```

Comment:
Can the axiosresponse be typed? I think `res.data` is a string in this case?

Reply:
the axiosresponse can be typed, yes. you wanna write that type?

Reply:
Can't it just be string? That generic on response is just the type of `.data`

Reply:
`(res: AxiosResponse<string>[]) => sortByPoints(res.map(c => parse(c.data)))`